plugins/web-terminal/src/components/NamespaceModalComponent/NamespaceModalComponent.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  FormControl,
+  Select,
+  MenuItem,
+  makeStyles,
+  Button,
+} from '@material-ui/core';
+import React, { useEffect } from 'react';
+import { getNamespaces } from '../TerminalComponent/utils';
+
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'center',
+    marginBottom: '10px',
+  },
+  buttonmargin: {
+    marginTop: '10px',
+    marginBottom: '10px',
+  },
+});
+
+type props = {
+  token: string;
+  link: string;
+  open: boolean;
+  previousNamespace: string;
+  handleClose: () => void;
+  onSubmit: (selectedNamesapce: string) => void;
+};
+
+export const NamespaceModalComponent = ({
+  token,
+  link,
+  open,
+  previousNamespace,
+  handleClose,
+  onSubmit,
+}: props) => {
+  const classes = useStyles();
+  const namespaceRef = React.useRef<HTMLInputElement>(null);
+  const [selectedNamespace, setSelectedNamespace] = React.useState('');
+  const [namespaces, setNamespaces] = React.useState([]);
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setSelectedNamespace(event.target.value as string);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onSubmit(selectedNamespace);
+  };
+
+  useEffect(() => {
+    getNamespaces(link, token).then(namespacesCluster => {
+      setNamespaces(namespacesCluster);
+    });
```

Comment:
`namespacesCluster` is a really weird name. Let's simplify this:

```suggestion
    getNamespaces(link, token).then(setNamespaces);
```

Reply:
Resolved